### PR TITLE
Update Doxygen config file and fix Doxygen warnings

### PR DESCRIPTION
**Pull request by LaszloLango**
A lot of warnings remained hibben because 'EXTRACT_ALL' was previously set to YES.

JerryScript-DCO-1.0-Signed-off-by: László Langó llango.u-szeged@partner.samsung.com

### CODE REVIEW

**Comment by LaszloLango** [comment]
@zherczeg there are some TODOs in this patch where I was not sure what would be the appropriate comment. Could you help me to fix them?

File: jerry-core/ecma/operations/ecma-exceptions.c [hunk 1]
```diff
@@ -34,12 +34,19 @@
  * \addtogroup exceptions Exceptions
  * @{
  */
+
+/**
+ * Type of mapping of errors.
```
Comment (zherczeg) [inline jerry-core/ecma/operations/ecma-exceptions.c]: Map error type to error prototype.

File: jerry-core/jmem/jmem-heap.c [hunk 2]
```diff
@@ -91,10 +95,15 @@
...
 #endif /* ECMA_VALUE_CAN_STORE_UINTPTR_VALUE_DIRECTLY */
+/**
+ * @}
+ */
 
 #ifndef JERRY_SYSTEM_ALLOCATOR
 /**
  * Get end of region
+ *
+ * @return pointer a free region node
```
Comment (zherczeg) [inline jerry-core/jmem/jmem-heap.c]: pointer to the end of the region

File: jerry-core/parser/js/js-lexer.c [hunk 3]
```diff
@@ -33,6 +33,9 @@
  * @{
  */
 
+/**
+ * Check whether the UTF-8 intermidate is an octet or not
```
Comment (zherczeg) [inline jerry-core/parser/js/js-lexer.c]: intermedate

File: jerry-core/parser/js/js-lexer.c [hunk 4]
```diff
@@ -971,6 +979,12 @@ lexer_parse_number (parser_context_t *context_p) /**< context */
   context_p->source_p = source_p;
 } /* lexer_parse_number */
 
+/**
+ * TODO
+ *
+ * @param char1 TODO
+ * @param type1 TODO
+ */
 #define LEXER_TYPE_A_TOKEN(char1, type1) \
```
Comment (zherczeg) [inline jerry-core/parser/js/js-lexer.c]: One character long token (e.g. comma).

File: jerry-core/parser/js/js-lexer.c [hunk 5]
```diff
@@ -979,6 +993,14 @@ lexer_parse_number (parser_context_t *context_p) /**< context */
...
   }
 
+/**
+ * TODO
+ *
+ * @param char1 TODO
+ * @param type1 TODO
+ * @param char2 TODO
+ * @param type2 TODO
+ */
```
Comment (zherczeg) [inline jerry-core/parser/js/js-lexer.c]: Token pair, where the first token is prefix of the second (e.g. % and %=).

File: jerry-core/parser/js/js-lexer.c [hunk 6]
```diff
@@ -994,6 +1016,16 @@ lexer_parse_number (parser_context_t *context_p) /**< context */
...
+/**
+ * TODO
+ *
+ * @param char1 TODO
+ * @param type1 TODO
+ * @param char2 TODO
+ * @param type2 TODO
+ * @param char3 TODO
+ * @param type3 TODO
+ */
```
Comment (zherczeg) [inline jerry-core/parser/js/js-lexer.c]: Three tokens, where the first is the prefix of the other two. (e.g. &, &&, &=)

File: jerry-core/parser/js/js-lexer.c [hunk 7]
```diff
@@ -1393,7 +1425,9 @@ lexer_process_char_literal (parser_context_t *context_p, /**< context */
   context_p->literal_count++;
 } /* lexer_process_char_literal */
 
-/* Maximum buffer size for identifiers which contains escape sequences. */
+/**
+ * Maximum buffer size for identifiers which contains escape sequences.
```
Comment (zherczeg) [inline jerry-core/parser/js/js-lexer.c]: Maximum local buffer size for ...

File: jerry-core/parser/js/js-lexer.c [hunk 8]
```diff
@@ -2113,11 +2147,17 @@ lexer_expect_identifier (parser_context_t *context_p, /**< context */
   parser_raise_error (context_p, PARSER_ERR_IDENTIFIER_EXPECTED);
 } /* lexer_expect_identifier */
 
+/**
+ * List of "get" locations
```
Comment (zherczeg) [inline jerry-core/parser/js/js-lexer.c]: Description of "get" literal string

File: jerry-core/parser/js/js-lexer.c [hunk 9]
```diff
@@ -2113,11 +2147,17 @@ lexer_expect_identifier (parser_context_t *context_p, /**< context */
...
+ */
 static const lexer_lit_location_t lexer_get_literal =
 {
   (const uint8_t *) "get", 3, LEXER_IDENT_LITERAL, false
 };
 
+/**
+ * List of "set" locations
```
Comment (zherczeg) [inline jerry-core/parser/js/js-lexer.c]: Description of "set" literal string

File: jerry-core/parser/js/js-parser-util.c [hunk 10]
```diff
@@ -72,6 +72,12 @@ parser_emit_two_bytes (parser_context_t *context_p, /**< context */
   }
 } /* parser_emit_two_bytes */
 
+/**
+ * TODO
+ *
+ * @param context_p TODO
+ * @param byte TODO
+ */
```
Comment (zherczeg) [inline jerry-core/parser/js/js-parser-util.c]: Append byte to the end of the current byte code stream.

File: jerry-core/parser/js/js-parser.c [hunk 11]
```diff
@@ -1361,6 +1361,12 @@ parse_print_final_cbc (ecma_compiled_code_t *compiled_code_p, /**< compiled code
 
 #endif /* PARSER_DUMP_BYTE_CODE */
 
+/**
+ * TODO
+ *
+ * @param page_p TODO
+ * @param offset TODO
+ */
```
Comment (zherczeg) [inline jerry-core/parser/js/js-parser.c]: Forward iterator: move to the next byte code

File: jerry-core/parser/js/js-parser.c [hunk 12]
```diff
@@ -1370,6 +1376,13 @@ parse_print_final_cbc (ecma_compiled_code_t *compiled_code_p, /**< compiled code
     } \
   } while (0)
 
+/**
+ * TODO
+ *
+ * @param page_p TODO
+ * @param offset TODO
+ * @param real_offset TODO
+ */
```
Comment (zherczeg) [inline jerry-core/parser/js/js-parser.c]: Forward iterator: move to the next byte code. Also updates the offset of the previous byte code.

File: jerry-core/vm/vm.c [hunk 13]
```diff
@@ -513,6 +514,11 @@ opfunc_construct (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
   frame_ctx_p->stack_top_p = stack_top_p;
 } /* opfunc_construct */
 
+/**
+ * TODO
+ *
+ * @param destination TODO
+ */
```
Comment (zherczeg) [inline jerry-core/vm/vm.c]: Read literal index from the byte code stream into destination.

File: jerry-core/vm/vm.c [hunk 14]
```diff
@@ -524,9 +530,16 @@ opfunc_construct (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
...
+ * TODO
+ *
+ * @param literal_index TODO
+ * @param target_value TODO
+ *
+ * TODO: For performance reasons, we define this as a macro.
  * When we are able to construct a function with similar speed,
- * we can remove this macro. */
+ * we can remove this macro.
+ */
```
Comment (zherczeg) [inline jerry-core/vm/vm.c]: Get literal value by literal index.

Comment (LaszloLango) [reply jerry-core/vm/vm.c]: Fixed.

**Comment by LaszloLango** [comment]
@zherczeg thanks for the review, I've updated the PR.

File: jerry-core/parser/js/js-lexer.c [hunk 15]
```diff
@@ -971,6 +979,12 @@ lexer_parse_number (parser_context_t *context_p) /**< context */
   context_p->source_p = source_p;
 } /* lexer_parse_number */
 
+/**
+ * One character long token (e.g. comma).
+ *
+ * @param char1 character
```
Comment (zherczeg) [inline jerry-core/parser/js/js-lexer.c]: Do we use at-param with macro functions?

Comment (LaszloLango) [reply jerry-core/parser/js/js-lexer.c]: Yes, we do. There are a few examples in the current master. Unfortunately we cannot use inline comments for macro functions parameters.

Comment (zherczeg) [reply jerry-core/parser/js/js-lexer.c]: Few examples does not make a rule. I found far less positive examples than negative, so I would rather remove these.

Comment (LaszloLango) [reply jerry-core/parser/js/js-lexer.c]: Doxygen will warn if I remove these.

Comment (zherczeg) [reply jerry-core/parser/js/js-lexer.c]: OK

**Comment by yichoi** [comment]
LGTM

**Pull request merged**